crates/rust-analyzer/src/global_state.rs

```diff
@@ -67,6 +67,7 @@ pub(crate) struct GlobalState {
     req_queue: ReqQueue,
     pub(crate) task_pool: Handle<TaskPool<Task>, Receiver<Task>>,
     pub(crate) loader: Handle<Box<dyn vfs::loader::Handle>, Receiver<vfs::loader::Message>>,
+    pub(crate) vfs_config_version: usize,
```

Comment:
I wonder if this bit of state belongs to `vfs::Handle`? That way, the client won't be able to mess-up version numbers. That is,

```
pub trait Handle: fmt::Debug {
    fn set_config(&mut self, config: Config) -> u32
```

Or maybe even `-> OpaqueConfigId`, if we go full proper software engineering.

Reply:
Hmm, the problem is that `Config` contains the version already, so set_config could only return the version from that...

Reply:
But does it need to contain a version? Seems like the client can set config without a version, and vfs can reply "ok, this is the ID for your config". As in, config is just data, it get's identity only when submitted to vfs for processing.

Reply:
We'd still need this field to keep track of the config version though.

Reply:
Yeah, that's true. Still feels more reliable if the vfs owns the incrementing counter, but, indeed, this doesn't make the code actually simpler, so lgtm as is!